### PR TITLE
Add system_metric function to GenericArtifact

### PR DESCRIPTION
**Pull request by hsubbaraj-spiral**
## Describe your changes and why you are making these changes
Adds the `system_metric` capability to GenericArtifacts. Creates a mixin SystemMetricMixin that both GenericArtifact and TableArtifact use.
## Related issue number (if any)
[Eng-2436](eng-2436-fix-regression-in-system-metrics)
## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [x] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: sdk/aqueduct/artifacts/generic_artifact.py [hunk 1]
```diff
@@ -90,3 +95,23 @@ def describe(self) -> None:
...
         print(json.dumps(readable_dict, sort_keys=False, indent=4))
+
+    def system_metric(
+        self, metric_name: str, lazy: bool = False
+    ) -> numeric_artifact.NumericArtifact:
+        """Creates a system metric that represents the given system information from the previous @op that ran on the table.
+
+        Args:
+            metric_name:
+                name of system metric to retrieve for the table.
```
Comment (kenxu95) [inline sdk/aqueduct/artifacts/generic_artifact.py:106]: Can we capitalize the first letter of each sentence? "Name of system metric..." I think that's what we do elsewhere in the codebase, and also what the google docstring style guide seems to indicate.

Essentially, let's treat docstrings as real english, with proper capitalization and periods.

Comment (kenxu95) [reply sdk/aqueduct/artifacts/generic_artifact.py:106]: This comment applies for all the other docstrings in this PR.

File: sdk/aqueduct/artifacts/system_metric.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    """A mixin class for the system_metric function. This is used by GenericArtifacts and TableArtifacts."""
+
+    def list_system_metrics(self) -> List[str]:
+        """Returns a list of all system metrics available on the table artifact.
+        These system metrics can be set via the invoking the system_metric() method the table.
+
+        Returns:
+            A list of available system metrics on a table
+        """
+        return list(SYSTEM_METRICS_INFO.keys())
```
Comment (kenxu95) [inline sdk/aqueduct/artifacts/system_metric.py:26]: I wonder if we should return the descriptions along with each system metric name? Essentially, keeping it a dictionary. That sees more useful to the user than just a list of system metric names.

File: sdk/aqueduct/artifacts/system_metric.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+    def system_metric_helper(
```
Comment (kenxu95) [inline sdk/aqueduct/artifacts/system_metric.py:28]: should we prepend a "_" to this name?

File: sdk/aqueduct/client.py [hunk 4]
```diff
@@ -444,7 +444,7 @@ def publish_flow(
         engine: Optional[str] = None,
         artifacts: Optional[Union[BaseArtifact, List[BaseArtifact]]] = None,
         metrics: Optional[List[NumericArtifact]] = None,
-        checks: Optional[List[BoolArtifact]] = None,
+        checks: Optional[List[bool_artifact.BoolArtifact]] = None,
```
Comment (kenxu95) [inline sdk/aqueduct/client.py:447]: Was this a necessary change?

**Comment by cw75** [comment]
@hsubbaraj-spiral how much work do you think it'll be to add the functionality to other artifact types?

**Review by kenxu95 — APPROVED** [review]

**Pull request merged**